submissions/misshatsker/object-oriented-js/app.js

```diff
@@ -0,0 +1,171 @@
+var leftSide = 0;
+var rightSide = 500;
+var topSide = 0;
+var bottomSide = 500;
+var initialGroundY = 60;
+
+var Enemy = function(x, y, speed) {
+    this.sprite = 'images/enemy-bug.png';
+    this.speed = speed;
+    this.x = x;
+    this.y = y;
+};
+
+Enemy.prototype.update = function(dt) {
+    var shift = this.speed * dt;
+    this.x = this.x + shift;
+    if (this.x >= rightSide) {
+        this.x = -this.width;
+    }
+};
+
+Enemy.prototype.render = function() {
+    ctx.drawImage(Resources.get(this.sprite), this.x, this.y);
+};
+
+Enemy.prototype.width = 80;
+Enemy.prototype.height = 70;
+
+var Player = function(x, y, speed) {
+    this.sprite = 'images/char-princess-girl.png';
+    this.speed = speed;
+    this.x = x;
+    this.y = y;
+};
+
+Player.prototype.width = 70;
+Player.prototype.height = 70;
+
+Player.prototype.update = function() {
+    const isOverlapDetected = allEnemies.some((enemy) => {
+        return checkIsOverlap(
+            {
+                x: player.x,
+                y: player.y
+            },
+            {
+                x: player.x + player.width,
+                y: player.y + player.height
```

Comment:
Why we refer to a specific `Player` instance inside class definition?
What if we have two players and rename that global variable?
Make the code safe.
Also requirements address this situation specifically.
Please find it, post here as a comment and fix code accordingly.

Also check other fragments of the code against this requirement and fix accordingly.